cli/cli_test.go

```diff
@@ -575,6 +575,33 @@ func Example_sql() {
 	// 0	empty	-
 }
 
+func Example_sql_escape() {
+	c := newCLITest()
+	defer c.stop()
+
+	c.RunWithArgs([]string{"sql", "-e", "create database t; create table t.t (s string);"})
+	// We insert four rows:
+	//  - a valid printable UTF8 string
+	//  - a valid printable UTF8 string using escapes
+	//  - a valid UTF8 string with a non-printable character
+	//  - a non-UTF8 string
+	c.RunWithArgs([]string{"sql", "-e", "insert into t.t values ('κόσμε'), (e'\\xc3\\xb1'), (e'\\x01'), (e'\\xc3\\x28')"})
```

Comment:
the values are encoded according to http://www.postgresql.org/docs/current/static/datatype-binary.html#AEN5667 so I think it would be `e'\\xe280ae'`, though when I tried that test locally I didn't get the kind of junk I was expecting given the output in #4315 (even without `%+q`).